src/asmcnc/skavaUI/screen_nudge.py

```diff
@@ -257,5 +252,22 @@ def generate_gcode():
         # Give time for wait popup to appear
         Clock.schedule_once(lambda dt: generate_gcode(), 0.5)
 
+    def set_datum_popup(self):
+        self.diff_x = (self.m.mpos_x() - self.initial_x)
+        self.diff_y = (self.m.mpos_y() - self.initial_y)
+
+        if abs(self.diff_x) > 3 or abs(self.diff_y) > 3:
+            # Display overall distance nudged to user, to 2dp
+            nudge_distance = "{:.2f}".format(math.hypot(self.diff_x, self.diff_y))
+            popup_nudge.PopupNudgeWarning(self.sm, self.m, self.l, nudge_distance)
+        else:
+            popup_nudge.PopupNudgeDatum(self.sm, self.m, self.l)
+
+    def set_datum(self):
+        # G54 and initial pos values do not get updated while on this screen, so nudges are always calculated from the same point
```

Comment:
I don't fully understand this comment/how this works? Are the g54 values not updated when the G10 command is written?